paddle/phi/infermeta/binary.cc

```diff
@@ -2510,6 +2510,24 @@ void MvInferMeta(const MetaTensor& x, const MetaTensor& vec, MetaTensor* out) {
   out->share_lod(x);
 }
 
+void PullSparseV2InferMeta(const MetaTensor& ids,
```

Comment:
```suggestion
void PullSparseV2InferMeta(const std::vector<const MetaTensor*>& ids,
```
现在的问题是这里的类型和签名里声明的算子类型不一致，对于vec<Tensor>类型的算子，它的infermeta要这样声明。对于ids和w，都要更改下类型。
输出out的类型要更改为
```
std::vector<MetaTensor*>
```

Reply:
收到